portability-api-launcher/src/main/java/org/dataportabilityproject/api/launcher/Logger.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 The Data-Portability Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dataportabilityproject.api.launcher;
+
+import java.util.function.Supplier;
+
+/**
+ * The system logger used to record events. Implementations may delegate to a logging implementation.
```

Comment:
Is this so that people don't have to use slf4j if they have a different implementation? If so, I feel like this may add more complexity than it's worth, I think it's OK to pick some impls of things like logging where there is a common, accepted library out there..

Plus different impls may not cleanly fit with this interface if the logging levels don't 100% map 1:1

Let me know if I'm misunderstanding what this is for

Reply:
What I was thinking here is an easy way for people to plug in their own log capture. For example, someone may want to have all logs sent directly to a cloud logging service. If we provide a direct mechanism where they pass in an instance, its easy for them to configure a logging service at startup before passing the instance to the core runtime. 

If we standardize on  SLF4J, someone will either have to pipe the output from an SLF4J implementation to the log service; create and SLF4J binder, or write a handler and set it via some configuration and rely on statics for configuration values.

Reply:
SGTM